src/commands/createFunction/durableSteps/ConfigureDurableProjectStep.ts

```diff
@@ -0,0 +1,127 @@
+import * as path from "path";
+import { Progress } from 'vscode';
+import { ConnectionKey, DurableBackend, hostFileName, ProjectLanguage } from '../../../constants';
+import { ext } from '../../../extensionVariables';
+import { IHostJsonV2 } from '../../../funcConfig/host';
+import { MismatchBehavior, setLocalAppSetting } from '../../../funcConfig/local.settings';
+import { localize } from '../../../localize';
+import { cpUtils } from '../../../utils/cpUtils';
+import { durableUtils, netheriteUtils, sqlUtils } from '../../../utils/durableUtils';
+import { pythonUtils } from '../../../utils/pythonUtils';
+import { venvUtils } from '../../../utils/venvUtils';
+import { IFunctionWizardContext } from '../IFunctionWizardContext';
+
+export class ConfigureDurableProjectStep<T extends IFunctionWizardContext> extends AzureWizardExecuteStep<T> {
+    public priority: number = 225;
+
+    public async execute(context: T, progress: Progress<{ message?: string; increment?: number }>): Promise<void> {
+        const configuring: string = localize('configuringDurableProject', 'Configuring durable project settings...');
+        progress.report({ message: configuring });
+
+        await this._configureHostAndLocalSettingsJson(context);
+        await this._installDurableDependencies(context);
+    }
+
+    public shouldExecute(context: T): boolean {
+        return !!context.newDurableStorageType;
+    }
+
+    private async _configureHostAndLocalSettingsJson(context: T): Promise<void> {
+        const hostJsonPath: string = path.join(context.projectPath, hostFileName);
+        if (!await AzExtFsExtra.pathExists(hostJsonPath)) {
+            const message: string = localize('failedToFindHost', 'Failed to find "{0}" in your project root.', hostFileName);
+            throw new Error(message);
+        }
+
+        const hostJson: IHostJsonV2 = await AzExtFsExtra.readJSON(hostJsonPath) as IHostJsonV2;
+        hostJson.extensions ??= {};
+
+        switch (context.newDurableStorageType) {
+            case DurableBackend.Storage:
+                hostJson.extensions.durableTask = durableUtils.getDefaultStorageTaskConfig();
+                break;
+            case DurableBackend.Netherite:
+                hostJson.extensions.durableTask = netheriteUtils.getDefaultNetheriteTaskConfig();
+                await setLocalAppSetting(context, context.projectPath, ConnectionKey.EventHub, '', MismatchBehavior.Overwrite);
+                break;
+            case DurableBackend.SQL:
+                hostJson.extensions.durableTask = sqlUtils.getDefaultSqlTaskConfig();
+                await setLocalAppSetting(context, context.projectPath, ConnectionKey.SQL, '', MismatchBehavior.Overwrite);
+                break;
+            default:
+        }
+
+        await AzExtFsExtra.writeJSON(hostJsonPath, hostJson);
+    }
+
+    private async _installDurableDependencies(context: T): Promise<void> {
```

Comment:
Bit of a nit: Since this can technically fail, it's nice to prefix this with try just so people are aware that its success is not guaranteed (or won't error out). Since it's a private function, it doesn't _really_ matter which is why it's more of a nit, but thought I'd point it out anyway.

```suggestion
    private async _tryInstallDurableDependencies(context: T): Promise<void> {
```